analyzers/src/SonarAnalyzer.Common/Common/NodeAndSymbol.cs

```diff
@@ -22,36 +22,41 @@
 
 namespace SonarAnalyzer.Common
 {
-    public class SyntaxNodeWithSymbol<TSyntaxNode, TSymbol>
+    public class NodeAndSymbol<TSyntaxNode, TSymbol>
         where TSyntaxNode : SyntaxNode
         where TSymbol : ISymbol
     {
-        public SyntaxNodeWithSymbol(TSyntaxNode syntax, TSymbol symbol)
+        public TSyntaxNode Node { get; }
+        public TSymbol Symbol { get; }
+
+        public NodeAndSymbol(TSyntaxNode node, TSymbol symbol)
         {
-            Syntax = syntax;
+            Node = node;
             Symbol = symbol;
         }
+    }
 
-        public TSyntaxNode Syntax { get; }
-        public TSymbol Symbol { get; }
+    public class NodeAndSymbol : NodeAndSymbol<SyntaxNode, ISymbol>
+    {
+        public NodeAndSymbol(SyntaxNode node, ISymbol symbol) : base(node, symbol) { }
     }
 
-    public static class SyntaxNodeWithSymbolHelper
+    public static class NodeAndSymbolHelper
     {
-        public static SyntaxNodeWithSymbol<TSyntaxNode, TSymbol> ToSyntaxWithSymbol<TSyntaxNode, TSymbol>(
+        public static NodeAndSymbol<TSyntaxNode, TSymbol> ToSyntaxWithSymbol<TSyntaxNode, TSymbol>(
             this TSyntaxNode syntax, TSymbol symbol)
             where TSyntaxNode : SyntaxNode
             where TSymbol : ISymbol
         {
-            return new SyntaxNodeWithSymbol<TSyntaxNode, TSymbol>(syntax, symbol);
+            return new NodeAndSymbol<TSyntaxNode, TSymbol>(syntax, symbol);
```

Comment:
May be we can use new return expression syntax here?

Reply:
I've removed it completely as well, it was too much code to do a simple thing (2 out of 3 usages didn't need the generic version).